tests/e2e/e2e_test.go

```diff
@@ -195,13 +234,29 @@ func (s *IntegrationTestSuite) TestTWAP() {
 	// Make sure that the pruning keep period has passed.
 	s.T().Logf("waiting for pruning keep period of (%.f) seconds to pass", initialization.TWAPPruningKeepPeriod.Seconds())
 	<-keepPeriodCountDown.C
-	oldEpochNumber := chainANode.QueryCurrentEpoch(epochIdentifier)
-	// The pruning should happen at the next epoch.
-	chainANode.WaitUntil(func(_ coretypes.SyncInfo) bool {
-		newEpochNumber := chainANode.QueryCurrentEpoch(epochIdentifier)
-		s.T().Logf("Current epoch number is (%d), waiting to reach (%d)", newEpochNumber, oldEpochNumber+1)
-		return newEpochNumber > oldEpochNumber
-	})
+
+	chainA.WaitForNumEpochs(1, epochIdentifier)
+
+	// We should not be able to get TWAP before swap since it should have been pruned.
+	s.T().Log("pruning is now complete, querying TWAP for period that should be pruned")
+	_, err = chainANode.QueryGetArithmeticTwapToNow(poolId, denomOne, denomTwo, timeBeforeSwap)
+	s.Require().Error(err)
+	s.Require().ErrorContains(err, "too old")
+
+	// TWAPs for the same time range should be the same when we query for them before and after pruning.
+	s.T().Log("querying for TWAP for period before pruning took place but should not have been pruned")
+	twapFromAfterToAfterSwapAndAfterPruning, err := chainANode.QueryGetArithmeticTwap(poolId, denomOne, denomTwo, timeAfterSwap, timeAfterSwap.Add(10*time.Millisecond))
+	s.Require().NoError(err)
+	s.Require().Equal(twapFromAfterToAfterSwapAndBeforePruning, twapFromAfterToAfterSwapAndAfterPruning)
+
+	// TWAP "from after to after swap" should equal to "from after swap to after pruning"
+	// That is because they are suppossed to use the same start and end records.
+	// The fact that these are taken over different time periods is irrelevant.
+	timeAfterPruning := chainANode.QueryLatestBlockTime()
+	s.T().Log("querying for TWAP from after swap to after pruning")
+	twapToNowPostPruning, err := chainANode.QueryGetArithmeticTwap(poolId, denomOne, denomTwo, timeAfterSwap, timeAfterPruning)
+	s.Require().NoError(err)
+	s.Require().Equal(twapToNowPostPruning, twapFromAfterToAfterSwapAndBeforePruning)
```

Comment:
This test sometimes fails and sometimes passes. However, I've verified through logs that it's grabbing the records from the desired heights that have correct spot prices.

The problem is likely coming from either time deltas or accumulator calculations when interpolating records.

@ValarDragon how do you think I should go about this? Is it worth investigating further or can we attribute this to setup/rounding errors? 

I might also be doing something wrong. Spent hours going in circles trying to root cause what exactly